Sources/SwiftSyntaxMacros/Syntax+LexicalContext.swift

```diff
@@ -92,6 +110,43 @@ extension SyntaxProtocol {
       if let parentContext = parentNode.asMacroLexicalContext() {
         parentContexts.append(parentContext)
       }
+      // Unfolded sequence expressions require special handling - effect marker
+      // nodes like `try`, `await`, and `unsafe` are treated as lexical contexts
+      // for all the nodes on their right. Cases where they don't end up
+      // covering nodes to their right in the folded tree are invalid and will
+      // be diagnosed by the compiler. This matches the compiler's ASTScope
+      // handling logic.
+      if let sequence = parentNode.as(SequenceExprSyntax.self) {
+        var sequenceExprContexts: [Syntax] = []
+        for elt in sequence.elements {
+          if elt.range.contains(self.position) {
+            // `sequenceExprContexts` is built from the top-down, but we
+            // build the rest of the contexts bottom-up. Reverse for
+            // consistency.
+            parentContexts += sequenceExprContexts.reversed()
```

Comment:
Wouldn’t it make more sense to move this after the `for` loop? Just to make sure that we add elements to `parentContext` in case we should terminate the loop for some other reason than the `break` below.

Reply:
Are there cases where we might not be able to determine which element the node is within? The main goal here is to only add the effects for nodes that occur after them in the sequence

Reply:
No, I just thought that it would read nicer to have 

```swift
            // `sequenceExprContexts` is built from the top-down, but we
            // build the rest of the contexts bottom-up. Reverse for
            // consistency.
            parentContexts += sequenceExprContexts.reversed()
```

after the `for` loop and only having a `break` in here because the addition of elements to `parentContexts` conceptually happens after iterating over the elements.

Reply:
Ah okay, I don't really feel all that strongly about it, happy to change